### PR TITLE
98679 - Added verify on tasks

### PR DESCRIPTION
**Pull request by arildbjerkenes**
[AB#98679](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/98679)

### CODE REVIEW

File: src/pages/Task/TaskSignature.tsx [hunk 1]
```diff
@@ -85,12 +131,40 @@ const TaskSignature = ({
...
+            ) : (
+                <>
+                    <p>This task is not verified.</p>
+                </>
+            )}
+
+            <ButtonWrapper>
+                <Button
+                    disabled={
+                        taskSignStatus === AsyncStatus.LOADING || isVerified
```
Comment (VRiveland) [inline src/pages/Task/TaskSignature.tsx:152]: In other pages the unsign/unclear button is hidden once the punch/checklist is verified (& verify button is hidden until the punch/checklist is signed/cleared) is there a reason for why this has been done differently here? If not, please change to match the punch/checklist page.

File: src/pages/Task/TaskSignature.tsx [hunk 2]
```diff
@@ -27,12 +39,17 @@ type TaskSignatureProps = {
...
     setIsSigned,
+    isVerified,
+    setIsVerified,
     task,
     setSnackbarText,
     refreshTask,
 }: TaskSignatureProps): JSX.Element => {
     const { api, params } = useCommonHooks();
     const [taskSignStatus, setTaskSignStatus] = useState(AsyncStatus.INACTIVE);
+    const [taskVerifyStatus, setTaskVerifyStatus] = useState(
```
Comment (VRiveland) [inline src/pages/Task/TaskSignature.tsx:50]: useState<AsyncStatus>(...
There are a few things we just need to fix whenever we see that they are missing. Adding a type for the useState is one of them

File: src/pages/Task/TaskSignature.tsx [hunk 3]
```diff
@@ -85,12 +131,40 @@ const TaskSignature = ({
...
                 </>
             )}
-            <Button
-                disabled={taskSignStatus === AsyncStatus.LOADING}
-                onClick={handleSign}
-            >
-                {isSigned ? 'Unsign' : 'Sign'}
-            </Button>
+
+            {task.verifiedAt ? (
```
Comment (VRiveland) [inline src/pages/Task/TaskSignature.tsx:135]: Is there a reason for this to be different to the checklist? If not, please don't show "This task is not verified", at least not before the task is signed.

File: src/pages/Task/Task.tsx [hunk 4]
```diff
@@ -149,6 +151,17 @@ const Task = (): JSX.Element => {
                     </Banner.Message>
                 </Banner>
             ) : null}
+            {isVerified ? (
```
Comment (VRiveland) [inline src/pages/Task/Task.tsx:154]: Please don't show both this and the banner for signed at the same time. Just show one banner, signed banner if signed but not verified, verified banner if verified. None if unsigned of course.

**Review by VRiveland — CHANGES_REQUESTED** [review]

File: src/pages/Task/TaskSignature.tsx (hunk 1 above)
Comment (arildbjerkenes) [reply src/pages/Task/TaskSignature.tsx:152]: Fixed

File: src/pages/Task/TaskSignature.tsx (hunk 3 above)
Comment (arildbjerkenes) [reply src/pages/Task/TaskSignature.tsx:135]: Did it like this because a saw that it was like this in procosys-main, but I agree...

**Pull request merged**